static/app/components/replays/breadcrumbs/timelineGaps.tsx

```diff
@@ -0,0 +1,82 @@
+import styled from '@emotion/styled';
+
+import * as Timeline from 'sentry/components/replays/breadcrumbs/timeline';
+import {getFramesByColumn} from 'sentry/components/replays/utils';
+import {
+  isBackgroundFrame,
+  isForegroundFrame,
+  type ReplayFrame,
+} from 'sentry/utils/replays/types';
+
+interface Props {
+  durationMs: number;
+  frames: ReplayFrame[];
+  width: number;
+}
+
+// create gaps in the timeline by finding all columns between a background frame and foreground frame
+// or background frame to end of replay
+export default function TimelineGaps({durationMs, frames, width}: Props) {
+  const markerWidth = frames.length < 200 ? 4 : frames.length < 500 ? 6 : 10;
+
+  const totalColumns = Math.floor(width / markerWidth);
+  const framesByCol = getFramesByColumn(
+    durationMs,
+    frames.filter(f => isBackgroundFrame(f) || isForegroundFrame(f)),
```

Comment:
This type of filter could be put into ReplayReader and memoized. `replayTimeline.tsx` could call something like `replay.getFooFrames()` to fetch just the list we need here.